### PR TITLE
New error calculation

### PR DESCRIPTION
**Pull request by jamesgk**
New error calculation. cc @behdad 
I figure the removal of error from return values will be controversial. cc @anthrotype
Note that the expected values in the unit tests are now from python3's random module. Also I think we can get further speedup by calculating error while calculating splines. So maybe this is still WIP.

### CODE REVIEW

**Comment by behdad** [comment]
Yes definitely calculate error while splitting.

File: Lib/cu2qu/__init__.py [hunk 1]
```diff
@@ -67,7 +67,7 @@ def dot(v1, v2):
 
 def lerp(a, b, t):
     """Linearly interpolate between scalars a and b at time t."""
-    return a * (1 - t) + b * t
+    return a + (b - a) * t
```
Comment (behdad) [inline Lib/cu2qu/__init__.py]: How much speedup if you inline the function?

Comment (jamesgk) [inline Lib/cu2qu/__init__.py]: How do you mean? Spelling it out everywhere? We call lerp a lot in the bezier_at functions so I don't think that would be worth the messy code.

Anyways, I think my estimate of 5-10% was a bit much. The benchmark results aren't all that consistent. But it seems generally to be a bit faster.

Comment (behdad) [inline Lib/cu2qu/__init__.py]: > How do you mean? Spelling it out everywhere? We call lerp a lot in the bezier_at functions so I don't think that would be worth the messy code.

Expand that code and you get a big speedup.  I should add those to fonttools...

**Comment by behdad** [comment]
I'll try to take a stab at this during my flight this weekend.

File: Lib/cu2qu/__init__.py (hunk 1 above)
Comment (behdad) [inline Lib/cu2qu/__init__.py]: Just did that and pushed that out.  With master, it resulted in a 2x speedup.

**Comment by anthrotype** [comment]
> I figure the removal of error from return values will be controversial

that was just for debugging, but I'm ok if you decide to drop it.
I'll need to some time to review #31 and #32, but don't wait for me -- I'll catch up as soon as I can.

**Comment by behdad** [comment]
Closing in preference for https://github.com/googlei18n/cu2qu/pull/33

**Pull request closed without merging**